### PR TITLE
Fix display of allowed statuses during creation

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #11858

### CODE REVIEW

File: templates/components/itilobject/fields/status.html.twig [hunk 1]
```diff
@@ -36,7 +36,8 @@
 {% if canupdate %}
    {% set status_field %}
       {{ item.dropdownStatus({
-         'value': item.fields['status'],
+         'value': item.input['status']|default(item.fields['status']),
+         'value_calculation': item.fields['status'],
```
Comment (cedric-anne) [inline templates/components/itilobject/fields/status.html.twig:40]: Unless I am wrong, `item.input` would be set only when `prepareInputFor...` returns false for a form submitted from UI, and, in this case, value was a valid value in the invalidated form. Not sure we have to introduce this `value_calculation`.

Comment (trasher) [reply templates/components/itilobject/fields/status.html.twig:40]: ping @cconard96

Comment (cconard96) [reply templates/components/itilobject/fields/status.html.twig:40]: The input array is filled when saved values are being restored after some form input refreshes the page.
That means the displayed value will try to be the new selection if it exists (input) or the item's current one (fields).
The value used to calculate which statuses are allowed, will always use the item's current one (fields).

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -3759,7 +3759,7 @@ public static function dropdownStatus(array $options = [])
 
         switch ($p['showtype']) {
             case 'allowed':
-                $tab = static::getAllowedStatusArray($p['value']);
+                $tab = static::getAllowedStatusArray($p['value_calculation'] ?? $p['value']);
```
Comment (VladoTTX) [inline src/CommonITILObject.php:3762]: Maybe change to:
``$tab = static::getAllowedStatusArray(empty($p['value_calculation'])?$p['value']:$p['value_calculation']);``
or update template is now passing also empty value during the item creation when fields['status'] does not exist:
'value_calculation': item.fields['status']

**Pull request merged**